modules/telemetry/src/specific/planetssonification.cpp

```diff
@@ -379,21 +378,98 @@ PlanetsSonification::PrecisionProperties::PrecisionProperties(
 {
     addProperty(lowDistancePrecision);
     lowDistancePrecision.setExponent(20.f);
-
     addProperty(highDistancePrecision);
     highDistancePrecision.setExponent(20.f);
-
     addProperty(lowAnglePrecision);
-
     addProperty(highAnglePrecision);
 }
 
+void PlanetsSonification::update(const Camera* camera) {
+    TelemetryModule* module = global::moduleEngine->module<TelemetryModule>();
+    if (!module) {
+        LERROR("Could not find the TelemetryModule");
+        return;
+    }
+
+    // The compare planets and planets overview sonifications depend on this
+    // sonification. If any of them are enabled, then this sonificaiton sends planetary
+    // data to the Open Sound Control receiver, even if this sonificaiton is disabled.
+    // @NOTE (malej 2024-11-28): Enabeling any combination of these three sonifications
+    // at the same time is not supported, only one will be used by SuperCollider
+    const TelemetryBase* compare = module->telemetry("PlanetsCompareSonification");
+    if (!compare) {
+        LERROR("Could not find the PlanetsCompareSonification");
+        return;
+    }
+    const TelemetryBase* overview = module->telemetry("PlanetsOverviewSonification");
+    if (!overview) {
+        LERROR("Could not find the PlanetsOverviewSonification");
+        return;
+    }
+
+    bool compareEnabled = compare->enabled();
+    bool overviewEnabled = overview->enabled();
+
+    if (!_enabled && !compareEnabled && !overviewEnabled) {
+        return;
+    }
+
+    // Get the angle settings from the module
+    TelemetryModule::AngleCalculationMode angleMode = module->angleCalculationMode();
+    bool includeElevation = module->includeElevationAngle();
+
+    const SceneGraphNode* focusNode =
+        global::navigationHandler->orbitalNavigator().anchorNode();
+
+    if (!focusNode) {
+        // The scene is likely not yet initialized
+        return;
+    }
+
+    // Update data for all planets
+    for (int i = 0; i < _planets.size(); ++i) {
+        // Increase presision if the planet is in focus
+        if (focusNode->identifier() == _planets[i].name) {
```

Comment:
```suggestion
    for (const DataBody& planet : _planets) {
        // Increase precision if the planet is in focus
        if (focusNode->identifier() == planet.name) {
```

Reply:
Same here, I need the `i` for the `updateData` function.